tests/metrics/collect_mem_consmd.sh

```diff
@@ -128,38 +134,115 @@ function start_collect()
 # It kills the collectl process.
 function stop_collect()
 {
+	echo "Stop collectl"
 	kill "$COLLECT_PID"
 }
 
-# Get the average about a "csv" file produced
-# by collectl tool.
-function get_average()
+# It launches a instance of psstop tool
+# in order to get metrics about proportional
+# set size.
+function start_psstop()
 {
+	echo "Launching psstop"
+	psstop_monitor=$(cat << EOF
+	#!/bin/bash
+
 	sum=0
 	count=0
 
+	while [ 1 ];do
+		if [ ! -f "$STOP_FILE" ];then
+			data=\$(psstop $PROC | awk '{print \$1}')
+			for i in \$data;do
+				if (( \$i > 0 ));then
+					sum=\$(( \$i + \$sum ))
+					count=\$(( \$count + 1))
+				fi
+			done
+
+			if (( \$count > 0 ));then
+				avg=\$(( \$sum / \$count ))
+				echo "\$avg" >> $PSS_FILE
+			fi
+
+			sleep 0.5
+		else
+			break;
+		fi
+	done
+EOF
+)
+	bash -c "$psstop_monitor" &
+	PSSTOP_PID=$!
+	echo "psstop pid: $PSSTOP_PID"
+}
+
+# It kills psstop process
+function stop_psstop()
+{
+	echo "Stop psstop" > "$STOP_FILE"
+}
+
+# Get PSS Average memory using psstop tool
+function get_psstop_average()
+{
+	pss_sum=0
+	pss_count=0
+
+	while read line; do
+		pss_mem=$(echo "$line" | sed 's/[A-Za-z]*//g')
+		if (( $pss_mem > 0 ));then
+			pss_sum=$(( $pss_sum + $pss_mem ))
+			pss_count=$(( $pss_count + 1 ))
+		fi
+	done < "$PSS_FILE"
+
+	average=$(( $pss_sum / $pss_count ))
```

Comment:
Again, this will give an integer result for `bash` (`zsh` doesn't have that limitation! :smile:)